src/utils/focus-management.ts

```diff
@@ -0,0 +1,179 @@
+// Credit: https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-vue/src/utils/focus-management.ts
+import { getDomElement } from '~/utils/dom'
+
+import type { Ref } from '@nuxtjs/composition-api'
+
+function getOwnerDocument<T extends Element | Ref<Element | null>>(
+  element: T | null | undefined
+) {
+  if (typeof window === 'undefined') return null
+  if (element instanceof Node) return element.ownerDocument
+  if (element && Object.prototype.hasOwnProperty.call(element, 'value')) {
+    const domElement = getDomElement(element)
+    if (domElement) return domElement.ownerDocument
+  }
+
+  return document
+}
+
+// Credit:
+//  - https://stackoverflow.com/a/30753870
+const focusableSelector = [
+  '[contentEditable=true]',
+  '[tabindex]',
+  'a[href]',
+  'area[href]',
+  'button:not([disabled])',
+  'iframe',
+  'input:not([disabled])',
+  'select:not([disabled])',
+  'textarea:not([disabled])',
+]
+  .map(
+    process.env.NODE_ENV === 'test'
+      ? // TODO: Remove this once JSDOM fixes the issue where an element that is
+        // "hidden" can be the document.activeElement, because this is not possible
+        // in real browsers.
+        (selector) =>
+          `${selector}:not([tabindex='-1']):not([style*='display: none'])`
+      : (selector) => `${selector}:not([tabindex='-1'])`
+  )
+  .join(',')
+
+export const Focus = Object.freeze({
+  /** Focus the first non-disabled element */
+  First: 1 << 0,
+
+  /** Focus the previous non-disabled element */
+  Previous: 1 << 1,
+
+  /** Focus the next non-disabled element */
+  Next: 1 << 2,
+
+  /** Focus the last non-disabled element */
+  Last: 1 << 3,
+
+  /** Wrap tab around */
+  WrapAround: 1 << 4,
+})
+
+export const FocusResult = Object.freeze({
+  Error: 1,
+  Overflow: 2,
+  Success: 4,
+  Underflow: 8,
+})
```

Comment:
Using this mapping to numbers has the strange effect of making the `focusIn` function's return type a `number`, which, at least to me, is pretty unclear. Switching this to be strings instead of numbers would make it clearer to a user of the function what the return type is meant to indicate, I think:

```suggestion
export const FocusResult = Object.freeze({
  Error: 'Error' as const,
  Overflow: 'Overflow' as const,
  Success: 'Success' as const,
  Underflow: 'Underflow' as const,
})
```

Then the return type of `focusIn` becomes `'Error' | 'Overflow' | 'Success' | 'Underflow'` which is much clearer than `number` (though we could still probably due to document what overflow and underflow mean as it is not immediately clear to me without reading the `focusIn` function's implementation carefully :slightly_smiling_face:)